add a release-blocking-jobs.md define a list of release-blocking jobs

krzyzacy opened the pull request:
this is the current state of the world for release-1.12, if we agree to this list then we can bikeshed other release dashboards.

Things on master-blocking but not release-blocking:
- conformance - openstack
- kubeadm
- package-pushd
- gce-scale-correctness
- gce-scale-performance

Things on older release-blocking dashboard
- kubeadm-x-on-y
- kubeadm-upgrade-x-to-y

(caveat on the difference: nobody was managing kubeadm jobs on 1.12 branch, also the gce-scale-* jobs are too expensive to run on each branch)

/sig testing
/sig release

(wonder where this file should live)
xref https://github.com/kubernetes/test-infra/issues/9363

cc @spiffxp @MaciekPytel @tpepper @AishSundar @foxish

neolit123 commented:
/cc @timothysc

neolit123 commented:
@krzyzacy 
> nobody was managing kubeadm jobs on 1.12 branch

could you please explain?
is this about the lack of kubeadm jobs here:
https://k8s-testgrid.appspot.com/sig-release-1.12-blocking

who's responsibility it is to add jobs to release blocking?
is it sig-release, or the release team or perhaps the project / some-area-of-k8s maintainers?

krzyzacy commented:
@hh did it for 1.11, ref https://github.com/kubernetes/test-infra/pull/8142

the process is not documented, I would expect sig-cluster-lifecycle to manage those jobs, rather than sig-release?

krzyzacy commented:
ack - sig-release didn't request that because we didn't have this list of job, that's why I added this doc :-)

neolit123 commented:
> the doc for kubeadm at least is WIP and i need to finishing it.

https://github.com/kubernetes/kubeadm/pull/1138

AishSundar commented on release-blocking-jobs.md:
```diff
@@ -0,0 +1,50 @@
+Here is a list of jobs that should block release if not healthy.
+
+If you want to add a release-branch job, please add them to sig-release-x.x-all dashboard first,
+and show the test is not flaky, then the release team can decide if the test should be added
```
I know @spiffxp was working on test flakiness criteria before letting jobs block a presubmit or release. May be worth linking to that here. If thats not ready yet, may be indicate a minimum threshold of stability before we allow a job to be release blocking

spiffxp replied:
I am fine with us merging as is and refining the criteria later

krzyzacy commented on release-blocking-jobs.md:
```diff
@@ -0,0 +1,50 @@
+  - kops-aws
+* Node
+  - node-kubelet
+* Kubeadm
```
why release team cares about a sig dashboard? We probably should link to @neolit123 's doc once that's done?

AishSundar replied:
if latest kubeadm tests are not integrated into the release-blockig dashboard (as with 1.12) there should be a way for someone to know where these results exists. I am ok linking to @neolit123 's doc, but from what I read even that seems to be WIP.

neolit123 replied:
i'm not sure what's best for this doc, but:

the WIP doc for kubeadm is going to be merged soon:
https://github.com/kubernetes/kubeadm/pull/1138

this PR for adding 1.12 branch kubeadm jobs is pending too:
https://github.com/kubernetes/test-infra/pull/9529

spiffxp commented on release-blocking-jobs.md:
```diff
@@ -0,0 +1,50 @@
+and show the test is not flaky, then the release team can decide if the test should be added
+to release-blocking dashboard as well.
+
+* Make & Unit & Integration
```
Meta-concern: this is a doc that can fall out of date with the testgrid dashboards.  I would rather we point to one of the dashboards as the authoritative source, reconcile or document the discrepancies, and thus keep the list in one place.

 ie:
- the release-1.12-blocking dashboard is the one true source
- all jobs in the dashboard should have descriptions
- it should be assumed that jobs are backported across all release branches unless otherwise noted in their description

krzyzacy replied:
My concern is, if we pin to one of the release dashboard, then what if we need to add more jobs?

I guess we can always use the master-blocking dashboard as the source of truth?

spiffxp replied:
If we need to add jobs, they can be backported, or their description can denote why they shouldn't be backported.

We're in this sticky place today where the master-blocking dashboard has some tests that will never show up on release-1.y-blocking dashboards, ie: scalability.  That's why I suggest using the latest release branch of kubernetes for now.  If we resolve that discrepancy before we cut another release branch, then we can use release-master blocking as the model.

Another approach we could shift to later: use a tool to generate release dashboards, then whatever source the tool uses becomes the source of truth.

spiffxp commented:
Don't block on me, we can reconcile docs later

krzyzacy commented:
added some bikesheds: 
- added a link to release-1.12-blocking dashboard
- as well as @neolit123 's kubeadm doc
- added TODO for switching to master-blocking, and define release-blocking criteria.

AishSundar commented:
/lgtm 
/hold

spiffxp commented:
/lgtm
as well

krzyzacy commented:
/hold cancel
let's iterate on future releases.
